src/graphql/root/query/on-chain-tx-fee-query.ts

```diff
@@ -0,0 +1,62 @@
+import { GT } from "@graphql/index"
+import * as Wallets from "@app/wallets"
+import * as Accounts from "@app/accounts"
+import WalletName from "@graphql/types/scalar/wallet-name"
+import OnChainAddress from "@graphql/types/scalar/on-chain-address"
+import SatAmount from "@graphql/types/scalar/sat-amount"
+import TargetConfirmations from "@graphql/types/scalar/target-confirmations"
+import OnChainTxFee from "@graphql/types/object/onchain-tx-fee"
+import { ApolloError, ForbiddenError } from "apollo-server-errors"
+
+const OnChainTxFeeQuery = GT.Field({
+  type: GT.NonNull(OnChainTxFee),
+  args: {
+    walletName: { type: WalletName },
+    address: { type: GT.NonNull(OnChainAddress) },
+    amount: { type: GT.NonNull(SatAmount) },
+    targetConfirmations: { type: TargetConfirmations, defaultValue: 3 },
+  },
+  resolve: async (_, args, { domainUser }) => {
+    const { walletName, address, amount, targetConfirmations } = args
+
+    for (const input of [walletName, address, amount, targetConfirmations]) {
+      if (input instanceof Error) throw input
+    }
+
+    let fee: Satoshis | Error | null = null
+    if (walletName) {
+      const hasPermissions = await Accounts.hasPermissions(domainUser.id, walletName)
+      if (hasPermissions instanceof Error) throw new ApolloError(hasPermissions.message)
+
+      if (!hasPermissions) throw new ForbiddenError("Invalid walletName")
```

Comment:
Can sending `Forbidden` here leak that the `walletName` existed? Shouldn't we return some kind of `NotFound`?

Reply:
we cant use normal not found.. I just returned an apollo error